Updated serverless quick start instructions (#13149)
DataDog/documentation: content/en/serverless/_index.md

{{< vimeo 543362476 >}}

 
<div class="alert alert-info">Make sure to check out discussions going on in the <a href="https://datadoghq.slack.com/archives/CFDPB83M4">#serverless</a> channel in the <a href="https://chat.datadoghq.com/">Datadog Slack community</a>.</div>

> [Datadog Serverless Monitoring][1] provides full visibility into all of the managed services that power your serverless applications by bringing together real-time metrics, logs and traces from your Lambda functions as well as related fully-managed APIs, queues, streams and data stores.

Reviewer: ```suggestion
[Datadog Serverless Monitoring][1] provides full visibility into all of the managed services that power your serverless applications by bringing together real-time metrics, logs, and traces from your Lambda functions as well as related fully-managed APIs, queues, streams, and data stores.
```

> [Datadog Serverless Monitoring][1] provides full visibility into all of the managed services that power your serverless applications by bringing together real-time metrics, logs and traces from your Lambda functions as well as related fully-managed APIs, queues, streams and data stores.

Reviewer: I'm wondering if the opening should be more agnostic with regards to serverless providers. we can certainly call out that most of this page concerns AWS Lambda, but maybe a link up top here to the Azure section + maybe a mention that GCP is cool too?

Author: I think a quick sentence relating to Azure or Google would make sense

Author: resolved

## Quick start

> If you are new to Datadog, you can get started by [signing up for a Datadog account][5], then following the instructions for [installing Datadog's Lambda Extension][6]. These steps configure your Lambda functions to send real-time metrics, logs, and traces to Datadog:

Reviewer 2: Should we also call out the link for existing datadog customers who are only new to serverless monitoring? (assuming we are going to add the lambda option on that page soon)

Author: we can revisit and it add it once its live

> If you are new to Datadog, you can get started by [signing up for a Datadog account][5], then following the instructions for [installing Datadog's Lambda Extension][6]. Completing these steps will configure your Lambda functions to send real-time metrics, logs and traces to Datadog:

Reviewer: ```suggestion
If you are new to Datadog, you can get started by [signing up for a Datadog account][5], then following the instructions for [installing Datadog's Lambda Extension][6]. These steps configure your Lambda functions to send real-time metrics, logs, and traces to Datadog:
```

{{< img src="serverless/aws-fully-managed-services-serverless-monitoring-hero.png" alt="Datadog Serverless Monitoring"  style="width:100%;" >}}

## Installation instructions

For more detailed installation instructions, select the Lambda runtime below for instructions to instrument your serverless application:

{{< partial name="serverless/getting-started-languages.html" >}}

> ## Explore Datadog Serverless Monitoring

Reviewer 2: Does it make sense to include the links to the corresponding blog posts here in case user has a strong interest in one of the use cases?

### Monitor your entire serverless stack in the Serverless view

> The Serverless view enables you to correlate high-level metrics from AWS resources with those of Lambda functions, so you can quickly spot issues and jump-start your investigation. 

Reviewer: ```suggestion
The Serverless view enables you to correlate high-level metrics from AWS resources with those of Lambda functions, so you can quickly spot issues and start your investigation. 
```

> By default, the Serverless view groups your serverless resources by service to help you easily visualize how each part of your application is performing. For each service, you will see the functions that belong to it, along with the resources (Amazon API Gateway, SNS, SQS, DynamoDB, S3, EventBridge, Kinesis) that invoked them.

Reviewer: ```suggestion
By default, the Serverless view groups your serverless resources by service to help you visualize how each part of your application is performing. For each service, you can see the functions that belong to it, along with the resources (Amazon API Gateway, SNS, SQS, DynamoDB, S3, EventBridge, Kinesis) that invoked them.
```

{{< img src="serverless/serverless-view-hero.jpeg" alt="Datadog Serverless Monitoring"  style="width:100%;" >}}

### Resolve AWS Lambda function failures faster by monitoring invocation payloads

> Datadog will automatically collect function requests and responses for all of your function invocations, providing key information that can help troubleshoot issues. For example, if you’re notified that one of your Lambda functions is experiencing failures, you can dive into the relevant request payloads to check for missing parameters, mistyped resource addresses, or other misconfigurations that may be behind the failures.

Reviewer: ```suggestion
Datadog automatically collects function requests and responses for all of your function invocations, providing key information that can help troubleshoot issues. For example, if you’re notified that one of your Lambda functions is experiencing failures, you can analyze relevant request payloads to check for missing parameters, mistyped resource addresses, or other misconfigurations that may be behind the failures.
```

By identifying misconfigurations in failing requests, you can more easily reproduce issues in your development environment—and then run tests to verify your bug fixes.

{{< img src="serverless/lambda_payload_hero.jpeg" alt="Datadog Serverless Monitoring"  style="width:100%;" >}}

### Real-time metrics for alerting on issues across your Lambda function environment

> Our enhanced Lambda metrics, which appear in Datadog with the prefix `aws.lambda.enhanced`, are available at down-to-the-second granularity and in near real time. Enhanced Lambda metrics can be used to alert or for SLOs on cold starts, estimated AWS costs, timeouts, out-of-memory errors and memory usage across all of your Lambda functions. This enables you to view performance issues in your serverless environments right as they occur and troubleshoot without delay. 

Reviewer: ```suggestion
Datadog's enhanced Lambda metrics, which appear in Datadog with the prefix `aws.lambda.enhanced`, are available at second granularity and in near real time. You can use enhanced Lambda metrics for alerts or SLOs on cold starts, estimated AWS costs, timeouts, out-of-memory errors, and memory usage across all of your Lambda functions. This enables you to view performance issues in your serverless environments as they occur and troubleshoot without delay. 
```
